teps/0135-coscheduling-pipelinerun-pods.md

```diff
@@ -393,8 +462,8 @@ E2E tests should be added, covering PipelineRuns where multiple PVC workspaces a
 
 As mentioned in ["Configuration"](#configuration), we'll need to replace the "disable-affinity-assistant" flag with the "coscheduling" flag. The steps are as follows:
 
-1. Coscheduling introduced as an alpha feature. Options for the "coscheduling" flag are "disabled" (default), "coschedule-pipelineruns", and "isolate-pipelineruns". The "disable-affinity-assistant" flag is announced as deprecated. If "disable-affinity-assistant" is set to "false", "coscheduling" must be set to "disabled", since the options for coscheduling are incompatible with the existing affinity assistant.
-1. After 9 months, or as part of a v1.0 software release (whichever happens first), the "disable-affinity-assistant" flag will be removed and replaced with a new "coscheduling" option "coschedule-workspaces" in the same release. The "coscheduling" flag will be set to "disabled" by default.
+1. Coscheduling introduced as an alpha feature. Options for the "coscheduling" flag are "coschedule-workspaces" (default), "disabled", "coschedule-pipelineruns", and "isolate-pipelineruns". The "disable-affinity-assistant" flag is announced as deprecated. If "disable-affinity-assistant" is set to "false", "coscheduling" must be set to "coschedule-workspaces", since the options for coscheduling are incompatible with the existing affinity assistant. When promiting this feature to beta, we may consider making "coschedule-pipelineruns" as the default value instead based on users' feedback.
+2. After 9 months, or as part of a v1.0 software release (whichever happens first), the "disable-affinity-assistant" flag will be removed and the affinity assistant behavior will be only determined by the "coscheduling" feature flags.
```

Comment:
Why is it that a `v1.0` release could shorten the deprecation timeframe for the flag?

Reply:
@lbernick , could you please help give more insights about `v1.0` release? (I'm not 100% sure as it is part of the original TEP 😅)

Reply:
my thought process was that we can introduce backwards incompatible changes in a v1.0.0 release, so it's sort of our "chance" to make any small backwards incompatible changes we want to make. The v1 TEP actually proposes swapping the default value of "disable-affinity-assistant" in v1. However, looking at semver.org more closely, this is not really what it says-- it says that before you release v1.0, "Anything MAY change at any time. The public API SHOULD NOT be considered stable" and afterwards, "Major version X (X.y.z | X > 0) MUST be incremented if any backward incompatible changes are introduced to the public API." (We aren't really following semver since we are trying to maintain backwards compatibility w/ version < 1.0.)

Anyway, I don't think this TEP is the right place to discuss semver and when it's appropriate to make backwards incompatible changes, so I think it would be fine to just say we'll remove the flag after an appropriate deprecation period, and have a separate discussion about what changes we want to make before releasing a v1.0.

Reply:
@QuanZhang-William this was marked as resolved but the 1.0 part is still in - do you plan on updating the text?

Reply:
Sorry @afrittoli , I missed this part, let me clean up the V1.